core/lib/libtomcrypt/src/mpa_desc.c

```diff
@@ -489,7 +489,9 @@ static int montgomery_reduce(void *a, void *b, void *c)
 	LTC_ARGCHK(b != NULL);
 	LTC_ARGCHK(c != NULL);
 	mpanum tmp;
-	init((void **)&tmp);
+	if (init((void **)&tmp) != CRYPT_OK) {
+		return CRYPT_MEM;
```

Comment:
Should not we return the error of `init()` instead (which is always CRYPT_MEM, I agree, but it looks cleaner). Same in other changes...